apps/web/src/components/WalletConnect/SessionProposalModal.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  Box,
+  Card,
+  FormControl,
+  FormErrorMessage,
+  HStack,
+  ModalContent,
+  Text,
+  VStack,
+  useToast,
+} from "@chakra-ui/react";
+import { type WalletKitTypes } from "@reown/walletkit";
+import { useDynamicModalContext } from "@umami/components";
+import { useAsyncActionHandler, useGetImplicitAccount, walletKit } from "@umami/state";
+import { buildApprovedNamespaces, getSdkError } from "@walletconnect/utils";
+import { useState } from "react";
+import { FormProvider, useForm } from "react-hook-form";
+
+import RequestModal from "./RequestModal";
+import { CheckmarkIcon, CloseIcon } from "../../assets/icons";
+import { OwnedImplicitAccountsAutocomplete } from "../AddressAutocomplete";
+
+/**
+ * Finds the overlap between the chains supported by the wallet and the chains requested by the dApp.
+ * We take the first required chain as Umami supports connection for only one chain per request.
+ */
+function getChainToConfirm(required: (string | string[] | undefined)[]) {
+  // take the first required chain which is supported by the wallet
+  for (const chain of required.flat()) {
+    if (chain && ["tezos:mainnet", "tezos:ghostnet"].includes(chain)) {
```

Comment:
We should probably use the actual list of networks here by calling `useAvailableNetworks()`

Also would be better to check by urls not by names - (mainnet & ghostnet are default, but for other networks user can select any name they like)

Could be a TODO for the following changes

Reply:
dApp in WalletConnect sends the names of the chains. How can they be selected by url? let's take it next steps

Reply:
If it's just `tezos:mainnet` & `tezos:ghostnet` - then there is no need for the check (for now, while we only support tezos). These two networks are always defined in Umami and can't be edited by users. We just need to decide on behaviour in the case of request with multiple chains

If there could be other network options we need to figure out how to check if they're present or not. 
Names for other (custom) networks are coming from the user and might not match WalletConnect names.

Reply:
it could be anything, but it must match the config on the wallet.
This is how the proposal looks like:

<img width="458" alt="image" src="https://github.com/user-attachments/assets/96d441a8-0b80-4326-90fb-b116582b38ec">